ci/azure-install-rust.yml

```diff
@@ -4,19 +4,18 @@ steps:
       if command -v rustup; then
         echo `command -v rustup` `rustup -V` already installed
         rustup self update
-      elif [ "$AGENT_OS" = "Windows_NT" ]; then
-        curl -sSf -o rustup-init.exe https://win.rustup.rs
-        rustup-init.exe -y --default-toolchain $TOOLCHAIN
-        echo "##vso[task.prependpath]$USERPROFILE/.cargo/bin"
       else
-        curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain $TOOLCHAIN
+        curl -sSL https://sh.rustup.rs | sh -s -- -y --default-toolchain="$TOOLCHAIN" --profile=minimal
         echo "##vso[task.prependpath]$HOME/.cargo/bin"
       fi
     displayName: Install rustup
 
   - bash: |
       set -e
-      rustup update $TOOLCHAIN
+      rustup set profile minimal
+      rustup component remove --toolchain=$TOOLCHAIN rust-docs || echo "already removed"
+      rustup component add --toolchain=$TOOLCHAIN rustc-dev || echo "already installed"
```

Comment:
Maybe move it *after* the call to `rustup update`?  I think it will fail otherwise, because nightly is not already installed, and "component add" won't be remembered.  Also, the echo "already installed" might be a little misleading, since it gobbles any error (such as doesn't exist, or download failure).

Also, maybe it should only install `rustc-dev` when using the nightly toolchain?  It won't be used for any of the others.  I'm also not sure if will be published for the stable toolchains?

Reply:
Ok. Switched the order and removed the echo. Now everything brakes. How can we install it only on nightly, I is not so experienced with azure?

Reply:
Oh, I was just thinking adding a conditional to the bash script, maybe something like:

```bash
if [ "$TOOLCHAIN" = "nightly" ]; then
  rustup component add --toolchain=$TOOLCHAIN rustc-dev
fi
```

just after `rustup update`.

You can also add it as a separate step with a [condition](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml), but I don't think it is necessary.